applets/wncklet/wayland-backend.c

```diff
@@ -457,24 +541,173 @@ static ToplevelTask *
 toplevel_task_new (TasklistManager *tasklist, struct zwlr_foreign_toplevel_handle_v1 *toplevel)
 {
 	ToplevelTask *task = g_new0 (ToplevelTask, 1);
+	GtkWidget *button;
+	GtkOrientation orient;
 
+	buttons = buttons + 1;
+	orient = gtk_orientable_get_orientation(GTK_ORIENTABLE(tasklist->outer_box));
 	task->button = gtk_button_new ();
 	g_signal_connect (task->button, "clicked", G_CALLBACK (toplevel_task_handle_clicked), task);
 
-	task->icon = gtk_image_new_from_icon_name ("unknown", GTK_ICON_SIZE_MENU);
+
+	task->icon = gtk_image_new_from_icon_name ("unknown", icon_size);
 
 	task->label = gtk_label_new ("");
 	gtk_label_set_width_chars (GTK_LABEL (task->label), -1);
 	gtk_label_set_ellipsize (GTK_LABEL (task->label), PANGO_ELLIPSIZE_END);
 	gtk_label_set_xalign (GTK_LABEL (task->label), 0.0);
 
 	GtkWidget *box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 0);
-	gtk_box_pack_start (GTK_BOX (box), task->icon, FALSE, FALSE, 2);
-	gtk_box_pack_start (GTK_BOX (box), task->label, TRUE, TRUE, 3);
+	gtk_box_pack_start (GTK_BOX (box), task->icon, FALSE, FALSE, 6);
+	gtk_box_pack_start (GTK_BOX (box), task->label, TRUE, TRUE, 2);
+
 	gtk_container_add (GTK_CONTAINER (task->button), box);
 	gtk_widget_set_name (task->button , "tasklist-button");
 	gtk_widget_show_all (task->button);
 
+	/* Buttons on a vertical panel are not affected by how many are needed
+	 * GTK handles compressing contents as needed as the window width tells
+	 * GTK how much space to allocate the label and icon. Buttons will use
+	 * the full width of a vertical panel without any special attention
+	 * so break out here instead of breaking the vertical panel case
+	 */
+
+	if (orient == GTK_ORIENTATION_VERTICAL)
+	{
+		gtk_widget_show_all (task->button);
+		task->toplevel = toplevel;
+		zwlr_foreign_toplevel_handle_v1_add_listener (toplevel,
+							&foreign_toplevel_handle_listener,
+							task);
+		g_object_set_data_full (G_OBJECT (task->button),
+							toplevel_task_key,
+							task,
+							(GDestroyNotify)toplevel_task_disconnected_from_widget);
```

Comment:
indents, too much tabs.
```suggestion
		g_object_set_data_full (G_OBJECT (task->button),
					toplevel_task_key,
					task,
					(GDestroyNotify)toplevel_task_disconnected_from_widget);
```

Reply:
This one was cut and pasted from the origjnal code, to keep the code below off the vertical panel case

Note that this whole file was done with tabs, so a reformat PR replacing tabs with spaces and doing nothing else may be called for. The tabs complicated figuring out how to handle this kind of spacing, as it is not consistant between different tab widths. Looks like you have 8 space tabs, and this may have been written with 8 space tabs orignally and pluma here defaults to 4 space tabs. Will reset to 8 space tabs for this formatting.

Between other PRs on this or after, we may want to do a PR to replace tabs with spaces and do nothing else.

Reply:
Done